### PR TITLE
fixes for the upcoming ECL release

### PR DESCRIPTION
**Pull request by spaghettisalat**
The upcoming ECL release includes a complete reimplementation of thread synchronization primitives. This allows us to support timeouts for locks as well as to retire all of the workarounds which implement timeouts using interrupts (which will not work in the next version).

### CODE REVIEW

**Comment by spaghettisalat** [comment]
Note that the new ECL version has now been released (https://ecl.common-lisp.dev/posts/ECL-2399-release.html).

**Comment by sionescu** [comment]
Please make the feature `has-timeouts` to avoid the double negation of `#-no-timeouts`.

**Comment by spaghettisalat** [comment]
> Please make the feature `has-timeouts` to avoid the double negation of `#-no-timeouts`.

Done

File: apiv1/impl-ecl.lisp [hunk 1]
```diff
@@ -8,8 +8,12 @@ Distributed under the MIT license (see LICENSE file)
 
 (in-package #:bordeaux-threads)
 
+(eval-when (:compile-toplevel :execute)
+  (unless (<= ext:+ecl-version-number+ 210201)
```
Comment (sionescu) [inline apiv1/impl-ecl.lisp:12]: This test should check that `+ecl-version-number+` is >= than the first known version to support timeouts. Which one is that ?

Comment (spaghettisalat) [reply apiv1/impl-ecl.lisp:12]: The version number is 230909

**Comment by sionescu** [comment]
Merged, thank you.

**Pull request closed without merging**